httpclient/src/main/kotlin/tbdex/sdk/httpclient/TbdexHttpClient.kt

```diff
@@ -70,34 +74,59 @@ object TbdexHttpClient {
   /**
    * Sends a message to the PFI.
    *
-   * @param message The [Message] object containing the message details to be sent.
-   * @throws TbdexResponseException for request or response errors.
+   * @param message The message to send.
+   *
+   * @throws TbdexResponseException for response errors.
    */
   fun sendMessage(message: Message) {
```

Comment:
@jiyoontbd what do you think about having a separate `fun createExchange(rfq: RFQ, replyTo: String?)` function for creating an exchange?

The reason being, there are now two functions that can be called to create an exchange. This one can accept messages of any type (including RFQ), but does not support `replyTo` URL. The other is also called `sendMessage`, but you can actually _only_ send an RFQ, not just any message type. 

You would essentially have one client method for creating an exchange, which can have a `replyTo` or not (optional), and a different method for all subsequent messages. 

Does that make sense?

Reply:
yea. so you're saying we'd call `createExchange()` for sending RFQ (with or without replyTo), and `sendMessage()` for everything else? i'll give that a go.

Reply:
i wish we could type guard for `sendMessage()` to prevent RFQ from being passed in as message in kotlin. so without the type guard, i'd have to check the `kind` and throw IllegalArgumentException. i'm not sure if i like that dev ex. i suppose we could explicitly say `pls don't use this method if you're sending an RFQ!!11` in the ktdocs. 

i'm leaning towards keeping the overloaded `sendMessage()`, would love to hear @diehuxx 's thoughts

Reply:
> i suppose we could explicitly say pls don't use this method if you're sending an RFQ!!11 in the ktdocs.

I would really prefer not to resort to that. IMO as long as we're leveraging the type system to enforce correct message types and options at compile-time, I can be flexible about how we name methods.

If I were writing this, I would keep the overloaded `sendMessage` AND add a `createExchange` endpoint which takes an RFQ and a `replyTo`. Admittedly, my ruby background is showing in this suggestion, as aliasing methods is common in ruby.

Reply:
i'm good with that. the code editor should show the difference between the two. [commit here](https://github.com/TBD54566975/tbdex-kt/pull/140/commits/2eed6181e9842762b497f9805d698b0faa6a863d#diff-3b01f04ac48b71c0a7b7ddaf971d8821e2849e72f1032dea144ea139684e7661R132-R141)